File: Doucment Editro/src/core/DocumentEditor.java
package core;

import model.Constants.ElementType;
import model.ElementTypes.AbsractDocumentElement;
import model.ElementTypes.ImageElementAbsract;
import model.ElementTypes.TextElementAbsract;

import java.util.ArrayList;
import java.util.List;

public class DocumentEditor {
    private final List<AbsractDocumentElement> elements = new ArrayList<>();

    public void addElement(AbsractDocumentElement element) {
        elements.add(element);
    }

    public List<AbsractDocumentElement> getElements() {
        return elements;
    }

    public void renderAll() {
        for (AbsractDocumentElement element : elements) {
            if (element.getType() == ElementType.TEXT) {
                TextElementAbsract textElement = (TextElementAbsract) element;
                System.out.println("Text: " + textElement.getText());
            } else if (element.getType() == ElementType.IMAGE) {
                ImageElementAbsract imageElement = (ImageElementAbsract) element;
                System.out.println("Image: " + imageElement.getImageUrl());
            }
        }
    }
}

File: 1) Document Editor/src/core/DocumentEditor.cpp
#include "core/DocumentEditor.h"
#include "models/TextElement.h"
#include "models/ImageElement.h"
#include <iostream>

using namespace std;

void DocumentEditor::addElement(DocumentElement* el) {
    elements.push_back(el);
}

void DocumentEditor::renderAll() const {
    for (DocumentElement* el : elements) {
        if (el->type == ElementType::Text) {
            TextElement* textEl = (TextElement*)el;
            cout << "Text: " << textEl->text << endl;
        } else if (el->type == ElementType::Image) {
            ImageElement* imageEl = (ImageElement*)el;
            cout << "Image: " << imageEl->imageUrl << endl;
        }
    }
}

DocumentEditor::~DocumentEditor() {
    for (DocumentElement* el : elements) {
        delete el;
    }
}
